src/OSSupport/GetAddressInfoError.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+
+
+/** Returns the readable form of a getaddressinfo type error code */
+inline AString ErrorString(int a_ErrorCode)
+{
+	// Note gai_strerror is not threadsafe on windows
+	#ifdef _WIN32
+		AString ErrorStr(GAI_STRERROR_BUFFER_SIZE, '\0');
+
+		int MsgLen = FormatMessageA(
+			FORMAT_MESSAGE_FROM_SYSTEM |
+			FORMAT_MESSAGE_IGNORE_INSERTS |
+			FORMAT_MESSAGE_MAX_WIDTH_MASK,
+			nullptr,
+			a_ErrorCode,
+			MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+			&ErrorStr[0],
```

Comment:
Is it really a hack? Looking at [cppreference](http://en.cppreference.com/w/cpp/string/basic_string) is says (emphasis mine):

>The elements of a basic_string are stored contiguously, that is, for a basic_string s, &*(s.begin() + n) == &*s.begin() + n for any n in [0, s.size()), or, equivalently, **a pointer to s[0] can be passed to functions that expect a pointer to the first element of a CharT[] array**. 

I'll change it if you wish but it seems like a good idea to save the extra copying.